### PR TITLE
feat: Introduce `resizable table` component

### PR DESCRIPTION
**Pull request by ironAiken2**
### TL;DR

Introduce BAITable, resizable table component.

### What changed?

- Implemented a table component that can be resized. Usage is the same as Table in traditional antd.

BAITable calculates and applies new widths based on a fixed width, so there are two rules to follow for smooth behavior
- BAITable **must have a width** specified in the column data. 
- If you use data that exceeds the width specified for the column, you must use the **ellipsis** property.

### How to test?

- Install the newly added packages. 
- Replace the parts of the webui that currently use table with BAITable.

---

**Checklist:** (if applicable)

- [ ] Mention to the original issue
- [ ] Documentation
- [ ] Minium required manager version
- [ ] Specific setting for review (eg., KB link, endpoint or how to setup)
- [ ] Minimum requirements to check during review
- [ ] Test case(s) to demonstrate the difference of before/after

### CODE REVIEW

**Comment by ironAiken2** [comment]
* **#2606** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2606?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ironAiken2 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by agatha197 — CHANGES_REQUESTED** [review]
Sorry for checking late. Please solve the conflicts

File: package.json [hunk 1]
```diff
@@ -123,6 +123,7 @@
     "@types/estree": "1.0.5",
     "@types/hammerjs": "^2.0.45",
     "@types/jest": "^29.5.12",
+    "@types/react-resizable": "^3.0.8",
```
Comment (yomybaby) [inline package.json]: You have to remove this because this is not a package.json of React project.

**Review by agatha197 — CHANGES_REQUESTED** [review]
If I set the width to overflow, that column looks like this.

But, when I added this code to the columns, you can avoid overflow. Can you generalize it?
```js
style={{
            overflow: 'hidden',
            whiteSpace: 'pre',
            wordWrap: 'break-word',
          }}
```

File: react/src/components/KeypairResourcePolicyList.tsx [hunk 2]
```diff
@@ -110,6 +110,7 @@ const KeypairResourcePolicyList: React.FC<KeypairResourcePolicyListProps> = (
       key: 'name',
       fixed: 'left',
       sorter: (a, b) => localeCompare(a?.name, b?.name),
+      width: 200,
```
Comment (agatha197) [inline react/src/components/KeypairResourcePolicyList.tsx]: Is it intended? If you want to reflect resizable property, making a child stack is better.

**Review by agatha197 — CHANGES_REQUESTED** [review]

File: react/src/components/KeypairResourcePolicyList.tsx (hunk 2 above)
Comment (ironAiken2) [reply react/src/components/KeypairResourcePolicyList.tsx]: I wrote the actual use case for ease of testing, but as you said, it seems better to have the application part on a separate stack. I'll comment that out.

File: react/src/components/KeypairResourcePolicyList.tsx [hunk 3]
```diff
@@ -110,6 +110,8 @@ const KeypairResourcePolicyList: React.FC<KeypairResourcePolicyListProps> = (
       key: 'name',
       fixed: 'left',
       sorter: (a, b) => localeCompare(a?.name, b?.name),
+      // to use resiazble table, width should be set e.g.)
```
Comment (agatha197) [inline react/src/components/KeypairResourcePolicyList.tsx]:
```suggestion
      // to use resizable table, width should be set e.g.)
```

**Review by agatha197 — CHANGES_REQUESTED** [review]
image
If I set the narrow width, I cannot see the part of the contents until widening that column.
By adding `whiteSpace: 'pre', wordWrap: 'break-word'` to https://github.com/lablup/backend.ai-webui/pull/2606/files#diff-20fe13a09e30cb99672ad684d2c592b13f7ba6c973a8f0ec7a208195e4e3edefR22  line, you can avoid it or you can add a new line or add ellipsis, etc.

**Review by agatha197 — CHANGES_REQUESTED** [review]
Please check https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2606/feat-Introduce-resizable-table-component#review-PRR_kwDOCRTcws6GsaqQ again

**Comment by ironAiken2** [comment]
Currently, the way vaadin tables show contents on resize is a mix of the `line wrapping` method you mentioned and the `overflow: hidden` method currently implemented.

The `overflow:hidden` method seems like a good idea because it doesn't break the table layout, but we'll have to talk about how to proceed. @agatha197 @yomybaby

File: react/src/components/KeypairResourcePolicyList.tsx [hunk 4]
```diff
@@ -110,6 +110,7 @@ const KeypairResourcePolicyList: React.FC<KeypairResourcePolicyListProps> = (
       key: 'name',
       fixed: 'left',
       sorter: (a, b) => localeCompare(a?.name, b?.name),
+      // width: 200,
```
Comment (yomybaby) [inline react/src/components/KeypairResourcePolicyList.tsx]: remove this line
```suggestion
```

**Review by yomybaby — CHANGES_REQUESTED** [review]
I just pushed two reviews to this PR but I fixed by myself and approve this PR. Additionally, I'll push new stack to add a feature and refactor BAITable. Please review and merge it. @ironAiken2

**Review by yomybaby — APPROVED** [review]
LGTM now. Please review my comments and the new stack, and merge all the stacks.

File: react/src/components/BAITable.tsx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+            return {
+              width: column.width,
+              onResize: handleResize(index) as React.ReactEventHandler<any>,
+            };
+          },
+        };
+        setTableColumns(newColumns);
+      };
+
+    setTableColumns(resizableColumns);
```
Comment (yomybaby) [inline react/src/components/BAITable.tsx:111]: @ironAiken2 Setting the state in the render phase is an anti-pattern. I removed this in #2748 . The `useMemo` callback is executed in the render phase.

**Review by agatha197 — APPROVED** [review]
LGTM
